### PR TITLE
[Batch] upgrade azure batch API version to 2024-02-01

### PR DESCRIPTION
**Pull request by wanghoppe**
**Related command**

**Description**

Upgrade azure batch API version to 2024-02-01
Upgrade `azure-batch` to 14.2.0
Upgrade `azure-mgmt-batch` to 17.3.0

**Testing Guide**

**History Notes**

[Batch] `az batch pool create` Add `--upgrade-policy-mode`, `--disable-auto-rollback`, `--enable-auto-os-upgrade`, `--defer-os-rolling-upgrade`, `--use-rolling-upgrade-policy`, `--enable-cross-zone-upgrade`, `--max-batch-instance-percent`, `-max-unhealthy-instance-percent`, `--max-unhealthy-upgraded-instance-percent`, `--pause-time-between-batches`, `--prioritize-unhealthy-instances`, and `--rollback-failed-instances-on-policy-breach` arguments to support automatic OS upgrade

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Batch

**Comment by wanghoppe** [comment]
Hi @zhoxing-ms, I saw you are the first assigned reviewers, and we'd like to include this change in the current milestone, could you review it as soon as you can? Thanks!

**Comment by zhoxing-ms** [comment]
> [Batch] az batch pool create add new parameters --upgrade-policy-mode, --disable-auto-rollback, --enable-auto-os-upgrade, --defer-os-rolling-upgrade, --use-rolling-upgrade-policy, --enable-cross-zone-upgrade, --max-batch-instance-percent, -max-unhealthy-instance-percent, --max-unhealthy-upgraded-instance-percent, --pause-time-between-batches, --prioritize-unhealthy-instances, and --rollback-failed-instances-on-policy-breach to support automatic OS upgrade

Please refer to this guideline [guideline_link](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests) to improve the `History Notes` in your PR description

File: src/azure-cli/azure/cli/command_modules/batch/_params.py [hunk 1]
```diff
@@ -281,6 +281,14 @@ def load_arguments(self, _):
...
                 help='Whether this pool should enable accelerated networking. Accelerated networking enables single root I/O virtualization (SR-IOV) to a VM, which may lead to improved networking performance. For more details, see: https://learn.microsoft.com/azure/virtual- network/accelerated-networking-overview. Set true to enable.')
+        c.argument('mode', options_list=['--upgrade-policy-mode'], help='The mode of the pool OS upgrade.')
+        c.argument('disable_automatic_rollback', options_list=['--disable-auto-rollback'], arg_type=get_three_state_flag())
+        c.argument('enable_automatic_os_upgrade', options_list=['--enable-auto-os-upgrade'], arg_type=get_three_state_flag())
+        c.argument('os_rolling_upgrade_deferral', options_list=['--defer-os-rolling-upgrade'], arg_type=get_three_state_flag())
+        c.argument('use_rolling_upgrade_policy', arg_type=get_three_state_flag())
+        c.argument('enable_cross_zone_upgrade', arg_type=get_three_state_flag())
+        c.argument('prioritize_unhealthy_instances', arg_type=get_three_state_flag())
+        c.argument('rollback_failed_instances_on_policy_breach', arg_type=get_three_state_flag())
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/batch/_params.py:291]: Why don't you add help messages for these new parameters?

Comment (wanghoppe) [reply src/azure-cli/azure/cli/command_modules/batch/_params.py:291]: I find those arguments have default help message from our python SDK which are already good enough so omitting here.

**Comment by wanghoppe** [comment]
> > [Batch] az batch pool create add new parameters --upgrade-policy-mode, --disable-auto-rollback, --enable-auto-os-upgrade, --defer-os-rolling-upgrade, --use-rolling-upgrade-policy, --enable-cross-zone-upgrade, --max-batch-instance-percent, -max-unhealthy-instance-percent, --max-unhealthy-upgraded-instance-percent, --pause-time-between-batches, --prioritize-unhealthy-instances, and --rollback-failed-instances-on-policy-breach to support automatic OS upgrade
> 
> Please refer to this guideline [guideline_link](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests) to improve the `History Notes` in your PR description

I capitalized the "Add" keywords according to the guide, not sure is there anything else needs to be change?

**Comment by zhoxing-ms** [comment]
No, what I mean is that you need to clearly describe each change in the PR description, so that customers can understand the specific functions of these changes. For example:
image

**Review by zhoxing-ms — APPROVED** [review]

**Review by necusjz — APPROVED** [review]

**Comment by evelyn-ys** [comment]
@wanghoppe Currently we are working on dropping all dependencies on `msrestazure` and we detected that the latest `azure-batch` still relies on `msrestazure` which is already deprecated.

Can you help remove `msrestazure` dependency from `azure-batch` and bump it in Azure CLI? Really appreciated.

**Pull request merged**